app/controllers/concerns/checkout_callbacks.rb

```diff
@@ -41,6 +41,12 @@ def load_saved_addresses
     @order.ship_address ||= finder.ship_address
   end
 
+  def load_saved_credit_cards
+    @saved_credit_cards = [] if !spree_current_user
+    @saved_credit_cards = spree_current_user.credit_cards.with_payment_profile.all if spree_current_user
```

Comment:
Since you are giving this so much thought, we could get rid of branches all together.
```suggestion
    @saved_credit_cards = spree_current_user&.credit_cards&.with_payment_profile.to_a
```

Reply:
`spree_current_user&.credit_cards&.with_payment_profile.to_a` returns an array by default? Ruby is quite amazing (but the truth is, I don't know if I can get used to it)